dags/airflow_db_prune_logs_rotate.py

```diff
@@ -0,0 +1,140 @@
+    """
+    Retrieve the number of days to prune back from the Airflow database.
+
+    Args:
+        days_back_to_prune (int): Number of days to look back for pruning.
+
+    Returns:
+        int: The number of days to prune back.
+    """
+    prune_before_days = int(days_back_to_prune)
+    logger = LoggingMixin().log
+    logger.info(f"Pruning Airflow DB records older than {prune_before_days} days")
+    return prune_before_days
+
+
+@task(task_id="get_clean_before_timestamp")
+def get_clean_before_timestamp(prune_before_days: int):
+    """
+    Calculate the ISO8601 timestamp before which records should be pruned.
+
+    Args:
+        prune_before_days (int): Number of days to look back for pruning.
+
+    Returns:
+        str: ISO8601 formatted timestamp.
+    """
+    clean_before_timestamp = (
+        pendulum.now("America/Chicago") - pendulum.duration(days=prune_before_days)
+    ).to_iso8601_string()
+    logger = LoggingMixin().log
+    logger.info(
+        f"Pruning records before {pendulum.parse(clean_before_timestamp).to_datetime_string()} America/Chicago, ISO8601: {clean_before_timestamp}"
+    )
+    return clean_before_timestamp
+
+
+@task.bash(task_id="airflow_db_clean")
+def db_clean_bash(timestamp: str) -> str:
+    """
+    Generate the bash command to clean the Airflow database before a given timestamp.
+
+    Args:
+        timestamp (str): ISO8601 formatted timestamp.
+
+    Returns:
+        str: Bash command string.
+    """
+    cmd = f'airflow db clean --yes --clean-before-timestamp "{timestamp}"'
+    logger = LoggingMixin().log
+    logger.info(f"Running command: {cmd}")
+    return cmd
+
+
+@task.bash(task_id="airflow_log_file_cleanup")
+def log_file_cleanup_bash(day_interval: int) -> str:
+    """
+    Generate the bash command to delete Airflow log files older than a given number of days.
+
+    Args:
+        day_interval (int): Number of days; logs older than this will be deleted.
+
+    Returns:
+        str: Bash command string.
+    """
+    cmd = (
+        f'find /opt/airflow/logs/ -type f -name "*.log" -mtime +{day_interval} -delete'
+    )
+    logger = LoggingMixin().log
+    logger.info(f"Running command: {cmd}")
+    return cmd
+
+
+@task.bash(task_id="airflow_log_dir_cleanup")
+def log_dir_cleanup_bash() -> str:
+    """
+    Generate the bash command to delete empty directories in the Airflow logs directory.
+
+    Returns:
+        str: Bash command string.
+    """
+    cmd = "find /opt/airflow/logs/ -type d -empty -delete"
+    logger = LoggingMixin().log
+    logger.info(f"Running command: {cmd}")
+    return cmd
+
+
+@dag(
+    dag_id="airflow_purge_logs_prune_database",
+    default_args=default_task_args,
+    schedule_interval="0 0 * * *" if DEPLOYMENT_ENVIRONMENT == "production" else None,
+    tags=["repo:atd-airflow", "airflow", "maintenance"],
+    catchup=False,
+    params={"days_back_to_prune": Param(default=30, type="integer", minimum=15)},
```

Comment:
`minimum=15`

You can't pick a number of days less that 15 for which you want to keep. The idea being that we always will have two weeks and change worth of logs, even if someone goes in there and cleans up everything they could.